File: internal/models/podcast.go
package models

type Podcast struct {
	ID              string        `json:"id"`
	Title           string        `json:"title"`
	Images          PodcastImages `json:"images"`
	IsExclusive     bool          `json:"isExclusive"`
	Publisher       string        `json:"publisherName"`
	PublisherID     string        `json:"publisherId"`
	MediaType       string        `json:"mediaType"`
	Description     string        `json:"description"`
	CategoryID      string        `json:"categoryId"`
	CategoryName    string        `json:"categoryName"`
	HasFreeEpisodes bool          `json:"hasFreeEpisodes"`
	PlaySequence    string        `json:"playSequence"`
}

type PodcastImages struct {
	Default   string `json:"default"`
	Featured  string `json:"featured"`
	Thumbnail string `json:"thumbnail"`
	Wide      string `json:"wide"`
}

type PodcastsResponse struct {
	Podcasts []Podcast `json:"podcasts"`
	Total    int       `json:"total"`
	Page     int       `json:"page"`
	Limit    int       `json:"limit"`
}

type SearchParams struct {
	Search       string
	Title        string
	CategoryName string
	Page         int
	Limit        int
}

File: frontend-ui/src/types/podcast.ts
export interface PodcastImage {
  default: string;
  featured: string;
  thumbnail: string;
  wide: string;
}

export interface Podcast {
  id: string;
  title: string;
  images: PodcastImage;
  isExclusive: boolean;
  publisherName: string;
  publisherId: string;
  mediaType: string;
  description: string;
  categoryId: string;
  categoryName: string;
  hasFreeEpisodes: boolean;
  playSequence: string;
}

export interface PodcastsResponse {
  data: Podcast[];
  isLoading: boolean;
  isError: boolean;
  error: unknown;
}

export interface SearchParams {
  search?: string;
  title?: string;
  categoryName?: string;
  page?: number;
  limit?: number;
}

export interface PodcastListProps {
  podcasts: Podcast[];
  isLoading: boolean;
  isError: boolean;
  error?: {
    status?: number;
    message: string;
  } | null;
}